src/test/java/ro/developmentfactory/thedrones/service/DronePositionServiceImplTest.java

```diff
@@ -0,0 +1,267 @@
+    void moveForward_ValidMove_ShouldUpdatePosition() {
+        // Given
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        DroneStatusResponse response = dronePositionService.moveForward(drone.getIdDrone());
+
+        // Then
+        assertEquals(5, response.getCurrentPositionX());
+        assertEquals(6, response.getCurrentPositionY());
+        assertEquals(Direction.N, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @Test
+    @DisplayName("Move Forward: Invalid drone ID should throw exception")
+    void moveForward_InvalidDroneId_ShouldThrowException() {
+        // Given
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.empty());
+
+        // When // Then
+        assertThrows(EntityNotFoundException.class, () -> dronePositionService.moveForward(UUID.randomUUID()));
+    }
+
+
+    @Test
+    @DisplayName("Move Forward: Multiple valid moves should update position correctly")
+    void moveForward_MultipleMoves_ShouldUpdatePositionCorrectly() {
+        // Given
+        droneStatus.setCurrentPositionX(0);
+        droneStatus.setCurrentPositionY(0);
+        droneStatus.setFacingDirection(Direction.E);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        dronePositionService.moveForward(drone.getIdDrone()); // Move to (1, 0)
+        dronePositionService.moveForward(drone.getIdDrone()); // Move to (2, 0)
+
+        // Then
+        assertEquals(2, droneStatus.getCurrentPositionX());
+        assertEquals(0, droneStatus.getCurrentPositionY());
+        assertEquals(Direction.E, droneStatus.getFacingDirection());
+        assertEquals(2, drone.getCountMove());
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "N,E",
+            "E,S",
+            "S,W",
+            "W,N"
+    })
+    @DisplayName("Turning Direction: Valid turn should update direction correctly")
+    void turningDirection_ValidTurn_ShouldUpdateDirection(Direction initialDirection, Direction expectedDirection) {
+        // Given
+        droneStatus.setFacingDirection(initialDirection);
+        when(droneRepository.findById(any(UUID.class))).thenReturn(Optional.of(drone));
+
+        // When
+        DroneStatusResponse response = dronePositionService.turn(drone.getIdDrone(), TurnDirection.RIGHT);
+
+        // Then
+        assertEquals(expectedDirection, response.getFacingDirection());
+        verify(droneStatusRepository).save(droneStatus);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "0,0,E,1,0",
+            "0,0,N,0,1",
+            "1,1,W,0,1",
+            "1,1,S,1,0"
+    })
+    @DisplayName("Move Forward: Valid moves should update position correctly")
+    void moveForward_ValidMoves_ShouldUpdatePositionCorrectly(int startX, int startY, Direction direction, int expectedX, int expectedY) {
```

Comment:
nice test! you can also pass and assert `expectedDirection` to check that it only moved and not rotated but it's very optional